src/Symfony/Bundle/FrameworkBundle/Command/DotenvDebugCommand.php

```diff
@@ -0,0 +1,139 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Dotenv\Dotenv;
+
+/**
+ * A console command to debug current dotenv files with variables and values.
+ *
+ * @author Christopher Hertel <mail@christopher-hertel.de>
+ */
+final class DotenvDebugCommand extends Command
+{
+    protected static $defaultName = 'debug:dotenv';
+    protected static $defaultDescription = 'Lists all dotenv files with variables and values';
+
+    private $kernelEnvironment;
+    private $projectDirectory;
+
+    public function __construct(string $kernelEnvironment, string $projectDirectory)
+    {
+        $this->kernelEnvironment = $kernelEnvironment;
+        $this->projectDirectory = $projectDirectory;
+
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title('Dotenv Variables & Files');
+
+        $envFiles = $this->getEnvFiles();
+        $availableFiles = array_filter($envFiles, function (string $file) {
+            return is_file($this->getFilePath($file));
+        });
+
+        if (\in_array('.env.local.php', $availableFiles, true)) {
+            $io->warning('Due to existing dump file (.env.local.php) all other dotenv files are skipped.');
+        }
+
+        $io->section('Scanned Files (in descending priority)');
+        $io->listing(array_map(static function (string $envFile) use ($availableFiles) {
+            return \in_array($envFile, $availableFiles, true)
+                ? sprintf('<fg=green>✓</> %s', $envFile)
+                : sprintf('<fg=red>⨯</> %s', $envFile);
+        }, $envFiles));
+
+        $io->section('Variables');
+        $io->table(
+            array_merge(['Variable', 'Value'], $availableFiles),
+            $this->getVariables($availableFiles)
+        );
+
+        $io->comment('Note real values might be different between web and CLI.');
+
+        return 0;
+    }
+
+    private function getVariables(array $envFiles): array
+    {
+        if (!\array_key_exists('SYMFONY_DOTENV_VARS', $_SERVER)) {
+            throw new \RuntimeException('Symfony Dotenv not initialized.');
+        }
+
+        $vars = explode(',', $_SERVER['SYMFONY_DOTENV_VARS'] ?? '');
+        sort($vars);
+
+        $output = [];
+        $fileValues = [];
+        foreach ($vars as $var) {
+            $realValue = $_SERVER[$var];
+            $varDetails = [$var, $realValue];
+            foreach ($envFiles as $envFile) {
+                $file = $this->getFilePath($envFile);
+                if (!is_file($file)) {
+                    continue;
+                }
```

Comment:
can be removed. If it continues the table output would be broken, but it can never happen ;) IMHO any error at this point should bubble.